src/main/kotlin/agallipo/BoardingCardSorter.kt

```diff
@@ -1,114 +1,49 @@
 package agallipo
 
-import java.io.ObjectInputFilter.merge
+class BoardingCardSorter : Journeys() {
 
-interface BoardingCardSorter {
+    fun sortTickets(boardingCards : List<BoardingCard>) : List<journey> {
+        var auxJourneys =  mutableListOf<mutableJourney>()
+        var allJourneys = listOf<journey>()
 
-    //This function divides  all the boarding cards into lists of flight-paths, iterating over all the boarding cards,
-    //it first checks if the boarding card can be joined to any journey, if not, it creates a new element in the
-    //list of flighPaths. Then it looks over if any flight path can be joined
-
-    fun sortTickets(boardingCards: MutableList<BoardingCard?>): MutableList<MutableList<BoardingCard?>> {
-        var flightPaths = mutableListOf<MutableList<BoardingCard?>>()
         for (boardingCard in boardingCards) {
-            if (!matchFlightPath(boardingCard, flightPaths))
-                flightPaths.add(mutableListOf<BoardingCard?>(boardingCard))
-            checkJoinFlightPath(flightPaths)
+            if (!matchFlightPath(boardingCard, auxJourneys))
+                allJourneys = addJourney(auxJourneys, boardingCard)
+            checkJoinFlightPath(auxJourneys)
         }
-        /*for (flight in flightPaths){
-            checkJoinFlightPath(flightPaths)
-        }*/
-        return flightPaths
+        return allJourneys
```

Comment:
Do you know how to analyze the complexity of an algorithm? check the concept: Big O notation to try to guess which complexity your algorithm has (https://medium.com/nowports-tech/introducci%C3%B3n-a-big-o-notation-95ecca8bd866) If the result is not good, I would:
1. try to implement some unit tests cases of sortTickets function
   - https://www.geeksforgeeks.org/unit-testing-software-testing/
   - https://kotlinlang.org/docs/jvm-test-using-junit.html

3. try to improve your algorithm thinking of other data structures that make easier the task of searching on lists: https://kotlinlang.org/docs/collections-overview